src/main/java/edu/kit/hci/soli/service/CurrentUserArgumentResolverService.java

```diff
@@ -0,0 +1,34 @@
+package edu.kit.hci.soli.service;
+
+import edu.kit.hci.soli.domain.User;
+import org.springframework.core.MethodParameter;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import java.security.Principal;
+
+public class CurrentUserArgumentResolverService implements HandlerMethodArgumentResolver {
+
+    private final UserService userService;
+
+    public CurrentUserArgumentResolverService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        return parameter.getParameterAnnotation(AuthenticationPrincipal.class) != null
+                && parameter.getParameterType().equals(User.class);    }
+
+    @Override
+    public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
+        Principal principal = (Principal) webRequest.getUserPrincipal();
+        if (principal != null) {
+            return userService.resolveLoggedInUser(principal);
+        }
```

Comment:
This is likely to be executed before the LSM advice, meaning we could end up with a null user for fresh logins.
You should probably move `User` creation here.